rotate.go

```diff
@@ -364,19 +365,81 @@ func (r *Rotator) SetSecret(ctx context.Context, event map[string]string) error
 	}
 	curUsername, curPassword := r.ss.Credentials(curVals)
 
+	curCred := db.Credentials{
+		Username: curUsername,
+		Password: curPassword,
+	}
+
+	newCred := db.Credentials{
+		Username: newUsername,
+		Password: newPassword,
+	}
+
 	// Combine the current and new credentials. This is plumbed all the way down
 	// into the db.PassswordSetter implementation.
 	creds := db.NewPassword{
-		Current: db.Credentials{
-			Username: curUsername,
-			Password: curPassword,
-		},
-		New: db.Credentials{
-			Username: newUsername,
-			Password: newPassword,
-		},
+		Current: curCred,
+		New:     newCred,
+	}
+
+	// Check to see if DB is already set to Pending password.
+	// This can happen if there's a previous run that did not complete successfully.
+	// Treat this as if SetPassword has completed successfully.
+	log.Println("Verifying if DB is already set to AWSPENDING version of secret")
+	if err := r.db.VerifyPassword(ctx, creds); err == nil {
+		r.event.Receive(Event{
+			Name: EVENT_END_PASSWORD_ROTATION,
+			Step: "setSecret",
+			Time: r.startTime,
+		})
+		log.Println("DB is already set to AWSPENDING version of secret, no action")
+		return nil
+	}
+
+	// Verify that credentials are valid before attempting to update secrets
```

Comment:
Why? Let's enhance this comment to explain to future engineers why we're verifying the creds are set before they're set.

Reply:
Updated